src/util/pg.py

```diff
@@ -1621,3 +1622,78 @@ def create_id_sequence(cr, table, set_as_default=True):
                 table=table_sql,
             )
         )
+
+
+def update_table_from_dict(cr, table, columns, mapping, key_col="id"):
```

Comment:
Nitpicking on the function name.
```suggestion
def bulk_update_table(cr, table, columns, mapping, key_col="id"):
```

Reply:
Better, indeed.